akka-actor/src/main/scala/akka/dispatch/Dispatchers.scala

```diff
@@ -97,9 +115,27 @@ class Dispatchers(val settings: ActorSystem.Settings, val prerequisites: Dispatc
         // It doesn't matter if we create a dispatcher configurator that isn't used due to concurrent lookup.
         // That shouldn't happen often and in case it does the actual ExecutorService isn't
         // created until used, i.e. cheap.
-        val newConfigurator =
-          if (cachingConfig.hasPath(id)) configuratorFrom(config(id))
-          else throw new ConfigurationException(s"Dispatcher [$id] not configured")
+
+        val newConfigurator: MessageDispatcherConfigurator =
+          if (cachingConfig.hasPath(id)) {
+            val valueAtPath = cachingConfig.getValue(id)
+            valueAtPath.valueType() match {
+              case ConfigValueType.STRING =>
+                // a dispatcher key can be an alias of another dispatcher, if it is a string
+                // we treat that string value as the id of a dispatcher to lookup, it will be stored
+                // both under the actual id and the alias id in the 'dispatcherConfigurators' cache
+                val actualId = valueAtPath.unwrapped().asInstanceOf[String]
+                logger.debug("Dispatcher id [{}] is an alias, actual dispatcher will be [{}]", id, actualId)
+                lookupConfigurator(actualId)
```

Comment:
ok, it's recursive, then we might need to protect against cycles (some max depth?)

Reply:
or we rely on stack overflow since it's not tailrec

Reply:
Yeah, I'd expect it to not be deep in the general case and to overflow if user did a mistake.

Reply:
I'd prefer an explicit check even if it's just to be able to provide a proper error message. Either hard-code the max-depth or make it configurable. Not more than a two-line change for the hard-coded version I guess ;)